### PR TITLE
Cs/onse dhis exponential backoff

### PR DESCRIPTION
**Pull request by Charl1996**
This PR is created for initial review before merging into `nh/dhis2/ui` base branch (not `master` yet).

### Notes on my choice of exponential backoff implementation:
Various methods have been explored:
- Adding decorator with sleep()
- Celery exponential backoff
- "Manual" backoff implementation

#### Adding decorator with sleep() 
This method is the most "clean" in my opinion as it allows one to simply add a decorator to the specific function or method which should have a backoff. However, it blocks the running thread until sleep() is done and we have Celery...

#### Celery exponential backoff
Celery does implement exponential backoff, but Dimagi has a forked version of Celery which did not have that feature implemented yet (would have been great, though).

#### "Manual" backoff
This method was chosen because it utilise Celery tasks to schedule a new task in the exponential future, should it be necessary. The drawback for me is the fact that it is coupled to the function/method to backoff, since it has to know about it's own retry state.

### Testing
Still exploring ways to reliably unit test this...

### CODE REVIEW

File: custom/onse/tasks.py [hunk 1]
```diff
@@ -83,9 +87,54 @@ def update_facility_cases_from_dhis2_data_elements(
...
+    always_eager = hasattr(settings, "CELERY_TASK_ALWAYS_EAGER") and settings.CELERY_TASK_ALWAYS_EAGER
+    settings.CELERY_TASK_ALWAYS_EAGER = False
+    callback_task.apply_async(tuple(args), eta=on_date)
+    settings.CELERY_TASK_ALWAYS_EAGER = always_eager
+
+
+def _server_is_online(dhis2_server: ConnectionSettings) -> bool:
+    network_locality = urlparse(dhis2_server.url).netloc
+    domain = network_locality.split(':')[0]
+    return os.system('ping -c 1 ' + domain) == PING_SUCCESS_CODE
```
Comment (Charl1996) [inline custom/onse/tasks.py:130]: Is sending one packet reliable enough?

**Review by kaapstorm — APPROVED** [review]

File: custom/onse/tasks.py (hunk 1 above)
Comment (kaapstorm) [reply custom/onse/tasks.py:130]: :shrug: Best of three? I'm not sure whether you get a success code on one successful ping, or whether you need all three to be successful.

Comment (kaapstorm) [reply custom/onse/tasks.py:130]: ... Alternatively, you could avoid the system call by using a socket:

```python
    try:
        sock = socket.create_connection((domain, 80), timeout=5)
    except:
        return False
    sock.shutdown()
    sock.close()
    return True
```

Comment (Charl1996) [reply custom/onse/tasks.py]: @snopoke thanks for the reference!

**Comment by Charl1996** [comment]
I'm closing this PR to recreate one to master.

**Pull request closed without merging**